src/test/resources/query-tests.yaml

```diff
@@ -107,6 +107,226 @@ tests:
     query: null + .
     output: "[2,3]"
 
+  -
+    input: |
+      {
+        "a": {
+          "b": 1,
+          "c": 2,
+          "d": 3
+        }
+      }
+    query: .a | [.b, .c, .d]
+    output: |
+      [1, 2, 3]
+  -
+    input: |
+      { "bar": "a", "baz": {"bar": "b"}}
+    query: |
+      {
+        "foo" : .bar,
+        * : .
+      } | .baz
+    output: |
+      {"bar": "b"}
+  -
+    input: |
+      { "bar": "a", "baz": {"bar": "b"}}
+    query: |
+      .baz | {
+        "foo" : .bar,
+        * : .
+      }
+    output: |
+      {"foo": "b", "bar": "b"}
+
+  -
+    input: |
+      { "foo": {"a": "b"}}
+    query: |
+      .foo | {"x": "y", * : .}
+    output: |
+      {
+        "x": "y",
+        "a": "b"
+      }
+
+  - input:
+    query: |
+      { "a": "b" } | {
+                        "type" : "Anonymized-View",
+                        * : .
+                     }
+    output: |
+      {
+        "type": "Anonymized-View",
+         "a": "b"
+      }
+
+  -
+    input:
+    query: |
+      {
+        "bar": "baz",
+        "foo" :
+          { "a": "b" } | {
+                            "type" : "Anonymized-View",
+                            * : .
+                         }
+      }
+
+    output: |
+      {
+        "bar": "baz",
+        "foo" :
+          {
+            "type" : "Anonymized-View",
+             "a": "b"
+          }
+      }
+
+  - input:
+    query: |
+      1 == 3 | (. - 2)
+    output: true
```

Comment:
I think this shows that the operator precedence is wrong. It seems to me that the pipe should have the highest precedence, so that this becomes `(1 == 3) | (. - 2)` and gives an error.

The same with the `or` example below. I think it should be `(true or false) | { "a" : .}` producing `{ "a" : true}`

What do you think?

Reply:
> It seems to me that the pipe should have the highest precedence

You mean lowest precedence ,right? 

I can't remember what was the rationale behind keeping comparators and logical operators lower than the pipe operator, so for me it's ok to change it to lowest if it feels more natural.

Maybe we should have a doc page just listing the operator precedence in general as well.

Reply:
> You mean lowest precedence ,right?

Argh. Yes, lowest. Never remember the terminology. :-)

> for me it's ok to change it to lowest if it feels more natural.

It felt that way to me. I say we try it, and once we have some experience with the master branch we can keep it or change it.

> Maybe we should have a doc page just listing the operator precedence in general as well.

Yeah, good point!

Reply:
Ok, I made pipe operator the lowest precedence operator,

I created issue #128 to handle adding the documentation page with the operator precedence